setup.py

```diff
@@ -1,23 +1,34 @@
 from setuptools import setup, find_packages
-from codecs import open
 from pathlib import Path
-import os
 from shutil import copy
+from collections import defaultdict
 
-path = os.path.abspath(os.path.dirname(__file__))
+root = Path(__file__).parent
 
-with open(os.path.join(path, "README.md")) as f:
+with open(root / "README.md") as f:
     long_description = f.read()
-with open(os.path.join(path, "requirements-minimal.txt")) as f:
+with open(root / "requirements-minimal.txt") as f:
     install_requires = f.readlines()
-with open(os.path.join(path, "requirements-full.txt")) as f:
-    full_dependencies = f.readlines()
-with open(os.path.join(path, "requirements-rtd.txt")) as f:
+with open(root / "requirements-full.txt") as f:
+    full_dependencies = set(f.readlines())
+with open(root / "requirements-rtd.txt") as f:
     documentation_dependencies = f.readlines()
-with open(os.path.join(path, "requirements-testing.txt")) as f:
+with open(root / "requirements-testing.txt") as f:
     testing_suite_dependencies = f.readlines()
 
-extras_require = dict(full=full_dependencies, test=testing_suite_dependencies, docs=documentation_dependencies)
+extras_require = defaultdict(list)
+subpaths = (root / "src" / "neuroconv" / "datainterfaces").rglob("*/requirements.txt")
+for subpath in subpaths:
+    with open(subpath) as f:
+        subpath_requirements = f.readlines()
+    extras_require[subpath.parent.name].extend(subpath_requirements)
+    full_dependencies.update(subpath_requirements)
```

Comment:
With this approach, we eventually we wouldn't even need to maintain a separate `requirements-full.txt` file with duplicated info - just rely on this machinery to amalgamate all the sub requirements automatically.